pkg/server/plugin/nodeattestor/k8s/psat/psat_test.go

```diff
@@ -0,0 +1,573 @@
+}
+
+func (s *AttestorSuite) TestAttestFailsWithMalformedToken() {
+	s.requireAttestError(makeAttestRequest("FOO", "blah"),
+		"k8s-psat: unable to parse token")
+}
+
+func (s *AttestorSuite) TestAttestFailsIfClusterNotConfigured() {
+	s.requireAttestError(makeAttestRequest("CLUSTER", "blah"),
+		`k8s-psat: not configured for cluster "CLUSTER"`)
+}
+
+func (s *AttestorSuite) TestAttestFailsWithBadSignature() {
+	// sign a token and replace the signature
+	token := s.signToken(s.fooSigner, TokenData{})
+	parts := strings.Split(token, ".")
+	s.Require().Len(parts, 3)
+	parts[2] = "aaaa"
+	token = strings.Join(parts, ".")
+
+	s.requireAttestError(makeAttestRequest("FOO", token),
+		"unable to verify token")
+}
+
+func (s *AttestorSuite) TestAttestFailsWithInvalidIssuer() {
+	token, err := jwt.Signed(s.barSigner).CompactSerialize()
+	s.Require().NoError(err)
+	s.requireAttestError(makeAttestRequest("BAR", token), "invalid issuer claim")
+}
+
+func (s *AttestorSuite) TestAttestFailsWithMissingNamespaceClaim() {
+	token := s.signToken(s.fooSigner, TokenData{})
+	s.requireAttestError(makeAttestRequest("FOO", token), "token missing namespace claim")
+}
+
+func (s *AttestorSuite) TestAttestFailsWithMissingServiceAccountNameClaim() {
+	token := s.signToken(s.fooSigner,
+		TokenData{
+			namespace: "NAMESPACE",
+			podName:   "PODNAME",
+			podUID:    "PODUID",
+		})
+	s.requireAttestError(makeAttestRequest("FOO", token), "token missing service account name claim")
+}
+
+func (s *AttestorSuite) TestAttestFailsWithMissingPodNameClaim() {
+	token := s.signToken(s.fooSigner,
+		TokenData{
+			namespace:          "NAMESPACE",
+			serviceAccountName: "SERVICEACCOUNTNAME",
+			podUID:             "PODUID",
+		})
+	s.requireAttestError(makeAttestRequest("FOO", token), "token missing pod name claim")
+}
+
+func (s *AttestorSuite) TestAttestFailsWithMissingPodUIDClaim() {
+	token := s.signToken(s.fooSigner,
+		TokenData{
+			namespace:          "NAMESPACE",
+			serviceAccountName: "SERVICEACCOUNTNAME",
+			podName:            "PODNAME",
+		})
+	s.requireAttestError(makeAttestRequest("FOO", token), "token missing pod UID claim")
+}
+
+func (s *AttestorSuite) TestAttestFailsIfServiceAccountNotWhitelisted() {
+	token := s.signToken(s.fooSigner,
+		TokenData{
+			namespace:          "NAMESPACE",
+			serviceAccountName: "SERVICEACCOUNTNAME",
+			podName:            "PODNAME",
+			podUID:             "PODUID",
+		})
+	s.requireAttestError(makeAttestRequest("FOO", token), `"NAMESPACE:SERVICEACCOUNTNAME" is not a whitelisted service account`)
+}
+
+func (s *AttestorSuite) TestAttestFailsIfTokenSignatureCannotBeVerifiedByCluster() {
+	token := s.signToken(s.bazSigner, TokenData{
+		namespace:          "NAMESPACE",
+		serviceAccountName: "SERVICEACCOUNTNAME",
+		podName:            "PODNAME",
+		podUID:             "PODUID",
+	})
+	s.requireAttestError(makeAttestRequest("FOO", token), "k8s-psat: unable to verify token")
+}
+
+func (s *AttestorSuite) TestAttestSuccess() {
```

Comment:
we should add another test case that asserts failure when the token is expired (SAT tokens don't expire, so there isn't an existing test for this).